build_tools/benchmarks/benchmark_helper.py

```diff
@@ -0,0 +1,166 @@
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Miscellaneous tool to help work with benchmark suite and benchmark CI."""
+
+import pathlib
+import sys
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
+
+import argparse
+import json
+from typing import List
+
+from e2e_test_framework import serialization
+from e2e_test_artifacts import model_artifacts, iree_artifacts
+from e2e_test_framework.definitions import iree_definitions
+
+IREE_COMPILER_NAME = "iree-compile"
+
+
+def _dump_cmds_of_generation_config(
+    gen_config: iree_definitions.ModuleGenerationConfig,
+    root_path: pathlib.PurePath = pathlib.PurePath()):
+
+  imported_model = gen_config.imported_model
+  imported_model_path = iree_artifacts.get_imported_model_path(
+      imported_model=imported_model, root_path=root_path)
+  module_path = iree_artifacts.get_module_dir_path(
+      module_generation_config=gen_config,
+      root_path=root_path) / iree_artifacts.MODULE_FILENAME
+  compile_cmds = [
+      IREE_COMPILER_NAME,
+      str(imported_model_path), "-o",
+      str(module_path)
+  ]
+  compile_cmds += gen_config.materialize_compile_flags()
+
+  if imported_model.import_config.tool == iree_definitions.ImportTool.NONE:
+    import_cmds = ["(Source model is already in MLIR)"]
+  else:
+    source_model_path = model_artifacts.get_model_path(
+        model=imported_model.model, root_path=root_path)
+    import_cmds = [
+        imported_model.import_config.tool.value,
+        str(source_model_path), "-o",
+        str(imported_model_path)
+    ]
+    import_cmds += imported_model.import_config.materialize_import_flags(
+        model=imported_model.model)
+
+  # TODO(#12215): Print benchmark name to make them searchable by keywords.
+  # Insert a blank line after each command to help read with line wrap.
+  return [
+      f'Compile Module: {" ".join(compile_cmds)}', "",
+      f'Import Model: {" ".join(import_cmds)}', ""
+  ]
+
+
+def _dump_cmds_from_run_config(
+    run_config: iree_definitions.E2EModelRunConfig,
+    root_path: pathlib.PurePath = pathlib.PurePath()):
+
+  gen_config = run_config.module_generation_config
+  module_path = iree_artifacts.get_module_dir_path(
+      module_generation_config=gen_config,
+      root_path=root_path) / iree_artifacts.MODULE_FILENAME
+
+  run_cmds = [run_config.tool.value, f"--module={module_path}"]
+  run_cmds += run_config.materialize_run_flags()
+  # TODO(#12215): Include benchmark name to make them searchable by keywords.
+  # Insert a blank line after the command to help read with line wrap.
+  lines = [f'Run Module: {" ".join(run_cmds)}', ""]
+  lines += _dump_cmds_of_generation_config(gen_config=gen_config,
+                                           root_path=root_path)
+  return lines
+
+
+def _dump_cmds_handler(args: argparse.Namespace):
```

Comment:
Could consider listing the kwargs here. I think that will give you slightly better checking for if a name exists (you can't remove something from args and forget to remove it from here as easily)

Reply:
Done.